core/internal/monitor/monitor.go

```diff
@@ -334,16 +335,33 @@ func (sm *SystemMonitor) Start(git *spb.GitRepoRecord) {
 		return // Already started or paused
 	}
 
-	// Probe the environment and resource metadata.
+	sm.git = git
+
+	// Start collecting metrics.
+	if !sm.settings.IsDisableStats() && !sm.settings.IsDisableMachineInfo() {
+		sm.logger.Debug("monitor: starting")
+		for _, resource := range sm.resources {
+			sm.wg.Add(1)
+			go sm.monitorResource(resource)
```

Comment:
No `sm.wg.Done()`?

It's usually best to put the `Done()` and `Add()` near each other (if the `Done()` is in `monitorResource()`, it should be moved here instead).